codecov_auth/views/okta.py

```diff
@@ -48,31 +48,38 @@ def validate_id_token(iss: str, id_token: str) -> dict:
 auth = HTTPBasicAuth(settings.OKTA_OAUTH_CLIENT_ID, settings.OKTA_OAUTH_CLIENT_SECRET)
 
 
-class OktaLoginView(LoginMixin, View):
-    def _fetch_user_data(self, iss: str, code: str) -> Optional[Dict]:
+class OktaLoginView(LoginMixin, StateMixin, View):
+    service = "okta"
 
+    def _fetch_user_data(self, iss: str, code: str, state: str) -> Optional[Dict]:
         res = requests.post(
             f"{iss}/oauth2/v1/token",
             auth=auth,
             data={
                 "grant_type": "authorization_code",
                 "code": code,
                 "redirect_uri": settings.OKTA_OAUTH_REDIRECT_URL,
+                "state": state,
             },
         )
 
+        if not self.verify_state(state):
+            log.warning("Invalid state during Okta OAuth")
+            return None
+
```

Comment:
we already verify the state in `_perform_login()` before we call `_fetch_user_data()`. do we also need to verify it here? i suppose it wouldn't hurt

Reply:
Not if it is triggered from perform login, but i thought it would be good to verify on fetch user if the flow is ever initiated from there.